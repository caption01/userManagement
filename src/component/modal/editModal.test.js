import React, {Component} from 'react';
import EditModal from './edit.modal.component';
import { shallow } from 'enzyme'

describe('Testing EditModal', () => {

    let wrapper ;
    beforeEach(() => {
        const mockProps = {
            userData: [],
            userInfo: {
                key: '10',
                first_name: 'john',
                last_name: 'mayer',
                address: 'USA'
            },
            updateUserData: jest.fn(),
            setActiveModal: jest.fn()
        }
        wrapper = shallow(<EditModal {...mockProps} />)
    })

    it('expect render EditModal', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('testing handleOk function for null userInfo props', () => {
        const mockPropsNullFirstName = {
            userData: [],
            userInfo: {
                key: '10',
                first_name: '',
                last_name: 'mayer',
                address: 'USA'
            },
            updateUserData: jest.fn(),
            setActiveModal: jest.fn()
        }

        window.alert = jest.fn();

        const wrapperCaseFirsName = shallow(<EditModal {...mockPropsNullFirstName} />)
        wrapperCaseFirsName.instance().handleOk()
        expect(window.alert).toBeCalledWith('please full fill infomation')

        const mockPropsNullLastName = {
            userData: [],
            userInfo: {
                key: '10',
                first_name: '',
                last_name: 'mayer',
                address: 'USA'
            },
            updateUserData: jest.fn(),
            setActiveModal: jest.fn()
        }

        const wrapperCaseLastName = shallow(<EditModal {...mockPropsNullLastName} />)
        wrapperCaseLastName.instance().handleOk()
        expect(window.alert).toBeCalledWith('please full fill infomation')


        const mockPropsNullAddress = {
            userData: [],
            userInfo: {
                key: '10',
                first_name: '',
                last_name: 'mayer',
                address: 'USA'
            },
            updateUserData: jest.fn(),
            setActiveModal: jest.fn()
        }

        const wrapperCaseAddress = shallow(<EditModal {...mockPropsNullAddress} />)
        wrapperCaseAddress.instance().handleOk()
        expect(window.alert).toBeCalledWith('please full fill infomation')
    })
})