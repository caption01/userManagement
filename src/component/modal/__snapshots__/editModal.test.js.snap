// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing EditModal expect render EditModal 1`] = `
<div
  className="edit-modal-container"
>
  <Modal
    confirmLoading={false}
    footer={
      Array [
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
        >
          Return
        </Button>,
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
          type="primary"
        >
          Submit
        </Button>,
      ]
    }
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    title="Edit User Infomation"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <p>
      First_Name : 
      <Input
        allowClear={true}
        id="inputfirstname"
        onChange={[Function]}
        type="text"
        value="john"
      />
       
    </p>
    <p>
      Last_Name : 
      <Input
        allowClear={true}
        id="inputlastname"
        onChange={[Function]}
        type="text"
        value="mayer"
      />
       
    </p>
    <p>
      Address : 
      <Input
        allowClear={true}
        id="inputaddress"
        onChange={[Function]}
        type="text"
        value="USA"
      />
       
    </p>
  </Modal>
</div>
`;
